File: Top Interview Questions/28. Implement strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.length() == 0) {
            return 0;
        }
        
        if (haystack.length() == 0 || haystack.length() < needle.length()) {
            return -1;
        }
        
        for (int i = 0; i <= haystack.length() - needle.length(); ) {
            int count = 0;
            int j = 0;
            while (j < needle.length() && haystack.charAt(i) == needle.charAt(j)) {
                count++;
                i++;
                j++;
            }
            
            if (count == needle.length()) {
                return i - count;
            } else {
                i = i - count + 1;
            }
        }
        
        return -1;
    }
}

File: Top Interview Questions/Two Pointers/28. Implement strStr().py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(needle) == 0:
            return 0
        elif len(needle) > len(haystack):
            return -1
        
        i = 0
        while i <= len(haystack) - len(needle):
            count, j = 0, 0
            while j < len(needle) and haystack[i] == needle[j]:
                count += 1; i+= 1; j += 1
            if count == len(needle):
                return i - count
            else:
                i = i - count + 1
                
        return -1